### ISSUE TITLE
Installation fails because of dependency conflict in requirements.txt

### ISSUE THREAD

**Issue by daerSeebaer** [rakuri255/UltraSinger#242]
I just tried to install from source.

When I run the "linux_cpu.sh" script, it starts going through the list of modules but throws an error. Relevant lines:

```
ERROR: Cannot install -r requirements.txt (line 645) [...] and numpy==1.26.4 because the package versions have conflicting dependencies.

The conflict is caused by:
The user requested numpy==1.26.4
whisperx 3.3.1 depends on numpy>=2.0.2
```

For whisperx there is no verison number specified in the requirements.txt. WhisperX added numpy as a dependency less than two months ago, I guess that broke it: https://github.com/m-bain/whisperX/commit/b41ebd4871f9cf9cf9b86bb61b446b2f5c9af29a

**Comment by ahallais** [comment]
Same issue with linux_gpu.sh
INFO: pip is looking at multiple versions of absl-py to determine which version is compatible with other requirements. This could take a while.
ERROR: Cannot install -r requirements.txt (line 142), -r requirements.txt (line 146), -r requirements.txt (line 163), -r requirements.txt (line 190), -r requirements.txt (line 204), -r requirements.txt (line 208), -r requirements.txt (line 21), -r requirements.txt (line 232), -r requirements.txt (line 252), -r requirements.txt (line 266), -r requirements.txt (line 311), -r requirements.txt (line 313), -r requirements.txt (line 315), -r requirements.txt (line 317), -r requirements.txt (line 341), -r requirements.txt (line 361), -r requirements.txt (line 373), -r requirements.txt (line 384), -r requirements.txt (line 416), -r requirements.txt (line 418), -r requirements.txt (line 452), -r requirements.txt (line 470), -r requirements.txt (line 478), -r requirements.txt (line 511), -r requirements.txt (line 526), -r requirements.txt (line 532), -r requirements.txt (line 534), -r requirements.txt (line 585), -r requirements.txt (line 605), -r requirements.txt (line 645), -r requirements.txt (line 65), -r requirements.txt (line 67), -r requirements.txt (line 69) and numpy==1.26.4 because these package versions have conflicting dependencies.

The conflict is caused by:
    The user requested numpy==1.26.4
    asteroid-filterbanks 0.4.0 depends on numpy
    contourpy 1.2.1 depends on numpy>=1.20
    crepe 0.0.16 depends on numpy>=1.14.0
    ctranslate2 4.4.0 depends on numpy
    h5py 3.11.0 depends on numpy>=1.17.3
    hmmlearn 0.3.2 depends on numpy>=1.10
    imageio 2.34.2 depends on numpy
    keras-preprocessing 1.1.2 depends on numpy>=1.9.1
    librosa 0.9.2 depends on numpy>=1.17.0
    lightning 2.3.3 depends on numpy<3.0 and >=1.17.2
    matplotlib 3.10.0 depends on numpy>=1.23
    music21 9.3.0 depends on numpy
    numba 0.60.0 depends on numpy<2.1 and >=1.22
    onnxruntime 1.18.1 depends on numpy<2.0 and >=1.21.6
    openunmix 1.3.0 depends on numpy
    opt-einsum 3.3.0 depends on numpy>=1.7
    optuna 3.6.1 depends on numpy
    pandas 2.2.2 depends on numpy>=1.22.4; python_version < "3.11"
    pretty-midi 0.2.10 depends on numpy>=1.7.0
    pyannote-core 5.0.0 depends on numpy>=1.10.4
    pyannote-metrics 3.2.1 depends on numpy
    pytorch-lightning 2.3.3 depends on numpy>=1.17.2
    pytorch-metric-learning 2.6.0 depends on numpy<2.0
    resampy 0.2.2 depends on numpy>=1.10
    scikit-learn 1.5.1 depends on numpy>=1.19.5
    scipy 1.14.1 depends on numpy<2.3 and >=1.23.5
    speechbrain 1.0.0 depends on numpy
    tensorboard 2.10.1 depends on numpy>=1.12.0
    tensorboardx 2.6 depends on numpy
    tensorflow 2.10.0 depends on numpy>=1.20
    torchmetrics 1.4.0.post0 depends on numpy>1.20.0
    transformers 4.39.3 depends on numpy>=1.17
    whisperx 3.3.1 depends on numpy>=2.0.2

To fix this you could try to:
1. loosen the range of package versions you've specified
2. remove package versions to allow pip attempt to solve the dependency conflict

ERROR: ResolutionImpossible: for help visit https://pip.pypa.io/en/latest/topics/dependency-resolution/#dealing-with-dependency-conflicts
Looking in indexes: https://download.pytorch.org/whl/cu118

**Comment by ahallais** [comment]
If someone has time to do a proper PR, it can be fixed by changing in
requirement.txt 
line 645 
whisperx @ git+https://github.com/m-bain/whisperx.git
by 
whisperx @ git+https://github.com/m-bain/whisperx.git@44e8bf5bb658ad6e1a80585b987fae2d48b9662c#egg=whisperx

**Comment by rakuri255** [comment]
I currently fixing the installer and moving to uv.
The current solution has a broken dependencie chain and cant be installed anymore.
